pkgs/development/compilers/heptagon/default.nix

```
@@ -0,0 +1,49 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, makeWrapper
+, ocamlPackages
+}:
+
+stdenv.mkDerivation rec {
+  pname = "heptagon";
+  version = "1.05.00";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.inria.fr";
+    owner = "synchrone";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-b4O48MQT3Neh8a1Z5wRgS701w6XrwpsbSMprlqTT+CE=";
+  };
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  buildInputs = with ocamlPackages; [
+    ocaml
+    findlib
+    menhir
+    menhirLib
+    camlp4
+    ocamlbuild
+  ];
+
+  propagatedBuildInputs = with ocamlPackages; [
+    lablgtk
+    ocamlgraph
+  ];
+
```

Nitpick: files in lib have executable bit toggled. I think it would be better to remove it with something like this:
```suggestion

  postInstall = ''
    find -type f $out/lib/heptagon -exec chmod -x {} \;
  ''

```

---

Also the fact that these files are in `lib` and not in something like `share` also bugs me a bit :)

---

The executable bits are now disabled.